More minor grunt QOL changes as outlined in balance request.

EpicFatGamer opened the pull request:
I removed all armor values for hands/feet the unggoy harness had, and added it to natural armor. This should give unggoy the incredibly small advantage of keeping their hand and foot armor even when their harness is damaged, as well as very slightly stronger armor in those areas overall. The justification is that trying to cut those areas in melee would be difficult, as it's covered by barbs and is extremely hard. 
The needler change is to make more kigyar/grunts take the weapon, as it's inability to fit on a back slot along with covenant's limited storage made it so you couldn't store it without giving up a belt, which is almost downright mandatory due to the lack of backpacks.  


<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You find a README and example file in .\html\changelogs\ for further instructions.
-->

<!-- If you need a change log update the below. Other wise you should remove it-->
:cl: EpicFatGamerNuts
tweak: moved armor values to natural armor from harness
tweak: slightly buffed said armor values
tweak: needler can fit on back 
/:cl:

EpicFatGamer committed:
Update grunt.dm

EpicFatGamer committed:
Update grunt.dm

EpicFatGamer committed:
Update pistols.dm

EpicFatGamer committed:
Update grunt.dm

EpicFatGamer committed:
Add armor to natural armor, remove from harness

To add a cool little unique part of being an unggoy, I removed the armor values the harness gave to hands and feet, and then added it to natural armor with a small buff (10 to melee because unggoy arms and legs look hard as hell, 5 to everything else.) Based on how often hands and feet get hit in combat, this should barely effect anything besides not needing a repair kit to keep full armor values on arms and legs (where there's no visible armor anyways?)

EpicFatGamer committed:
added needler to back slot

no one takes needler on grunt/jackal, and a large reason is because you can't take a medical or martial belt with it, reducing your already limited storage.

EpicFatGamer committed:
needler can fit on back

no one takes needler, and a large reason is because it couldn't fit on back slot, making you unable to choose a medical/martial belt. With unggoy and kigyar's already limited storage, people didn't want to lose more. now it can fit on back!

X0-11 commented:
travis angery

caelaislinn commented on code/modules/halo/species_items/grunt.dm:
```diff
@@ -20,7 +20,7 @@
 	icon_state = "combatharness_minor"
 
-	body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|FEET|ARMS|HANDS //Essentially, the entire body besides the head
+	body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS||ARMS| //Essentially, the entire body besides the head
```
This should be `body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|ARMS`

caelaislinn commented on code/modules/halo/species_items/grunt.dm:
```diff
@@ -143,6 +143,7 @@
 	item_state_slots = list(slot_l_hand_str = "armor", slot_r_hand_str = "armor")
 	species_restricted = list("Unggoy")
 	armor = list(melee = 5, bullet = 5, laser = 5, energy = 5, bomb = 5, bio = 0, rad = 0)
+	
```
Try to avoid randomly adding or deleting newlines

caelaislinn commented on code/modules/halo/species_items/grunt.dm:
```diff
@@ -185,9 +186,9 @@
-	//totalshields = 100
-	//specials = list(/datum/armourspecials/shields/unggoy)
+	
+	body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS||ARMS|
```
This should be `body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|ARMS`

caelaislinn commented on code/modules/halo/species_items/grunt.dm:
```diff
@@ -201,6 +202,8 @@
 
 	armor = list(melee = 45, bullet = 45, laser = 20, energy = 20, bomb = 60, bio = 0, rad = 0)
+	
+	body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS||ARMS|
```
This should be `body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|ARMS`

caelaislinn commented on code/modules/halo/species_items/grunt.dm:
```diff
@@ -225,6 +228,8 @@
 	item_state = "combatharness_honour"
 	totalshields = 100 //Pretty much just a distinguishing feature.
+	
+	body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS||ARMS|
```
This should be `body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|ARMS`

caelaislinn reviewed (changes requested):
Please avoid commit messages like "Update grunt.dm" as it's uninformative and not helpful. Your commit messages like "Add armor to natural armor, remove from harness " are good. A commit message should briefly describe what the changes are. 

Also it looks like someone merged a travis failing PR that modified the globals. I've updated the global checksum in #2233 which will fix that.

EpicFatGamer committed:
issues notes by cael fixed, corresponding slight feet/hand buff

Fixed all issues Cael noted, and buffed hand/foot armor to correspond with XO grunt armor buffs, making sure natural armor keeps the slight buff I intended to some areas and the same for others.Before this fix, it actually had LESS armor then some of the harnesses, specifically bomb and energy.

caelaislinn commented on code/modules/halo/species_items/grunt.dm:
```diff
@@ -20,7 +20,7 @@
 	icon_state = "combatharness_minor"
 
-	body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|FEET|ARMS|HANDS //Essentially, the entire body besides the head
+	body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|ARMS| //Essentially, the entire body besides the head
```
Still not quite correct. It needs to be `body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|ARMS` note the bar at the end is removed. 

Note that setting it here applies this setting to all the unggoy armour, so you don't need to reapply it later in the file.

caelaislinn commented on code/modules/halo/species_items/grunt.dm:
```diff
@@ -172,9 +173,11 @@
+	
 	armor = list(melee = 45, bullet = 45, laser = 20, energy = 20, bomb = 60, bio = 0, rad = 0)
 
+	body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|ARMS|
```
I did some checking and this line can actually be deleted. For it to work it needs to be `body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|ARMS` but for future code maintenance it's better to remove it.

caelaislinn commented on code/modules/halo/species_items/grunt.dm:
```diff
@@ -185,9 +188,9 @@
-	//totalshields = 100
-	//specials = list(/datum/armourspecials/shields/unggoy)
+	
+	body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|ARMS|
```
This should be `body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|ARMS` but it can also be deleted.

caelaislinn commented on code/modules/halo/species_items/grunt.dm:
```diff
@@ -201,6 +204,8 @@
 
 	armor = list(melee = 45, bullet = 45, laser = 20, energy = 20, bomb = 60, bio = 0, rad = 0)
+	
+	body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|ARMS|
```
This should be `body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|ARMS` but it can also be deleted.

caelaislinn commented on code/modules/halo/species_items/grunt.dm:
```diff
@@ -214,6 +219,8 @@
 
 	armor = list(melee = 45, bullet = 45, laser = 20, energy = 20, bomb = 60, bio = 0, rad = 0)
 
+	body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|ARMS|
```
This should be `body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|ARMS` but it can also be deleted.

caelaislinn commented on code/modules/halo/species_items/grunt.dm:
```diff
@@ -225,6 +232,8 @@
 	item_state = "combatharness_honour"
 	totalshields = 100 //Pretty much just a distinguishing feature.
+	
+	body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|ARMS|
```
This should be `body_parts_covered = UPPER_TORSO|LOWER_TORSO|LEGS|ARMS` but it can also be deleted.

BDpuffy420 commented:
going to close this
